### PR TITLE
refactor CI

### PR DESCRIPTION
**Pull request by danieleades**
- refactors the CI scripts to remove the unmaintained 'actions-rs' actions
- adds dependabot config
- configures MSRV
- adds an MSRV check to CI

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,47 +5,42 @@ on:
...
     steps:
       - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - uses: dtolnay/rust-toolchain@nightly
         with:
-          toolchain: stable
           components: clippy
-          args: -- -D warnings
-      - uses: actions-rs/cargo@v1
+      - run: cargo clippy --all-targets
```
Comment (hrektts) [inline .github/workflows/ci.yml:36]: Can you add `-- -D warnings` option?

File: Cargo.toml [hunk 2]
```diff
@@ -13,6 +13,7 @@ keywords = ["encode", "decode", "serialize", "deserialize"]
 categories = ["encoding", "network-programming"]
 license = "MIT/Apache-2.0"
 edition = "2018"
+rust-version = "1.56.1"
```
Comment (hrektts) [inline Cargo.toml:16]: Is there a rationale for specifying this version?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -5,47 +5,42 @@ on:
...
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-      - uses: actions-rs/cargo@v1
-        with:
-          command: test
+      - uses: dtolnay/rust-toolchain@nightly
```
Comment (hrektts) [inline .github/workflows/ci.yml:15]: Does this need the `nightly`?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -5,47 +5,42 @@ on:
...
+      - run: cargo test
 
   format:
     name: Format
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - uses: dtolnay/rust-toolchain@nightly
```
Comment (hrektts) [inline .github/workflows/ci.yml:23]: Same as above.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -5,47 +5,42 @@ on:
...
-          args: -- --check
+      - run: cargo fmt -- --check
 
   clippy:
     name: Clippy
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - uses: dtolnay/rust-toolchain@nightly
```
Comment (hrektts) [inline .github/workflows/ci.yml:33]: Same as above.

**Review by hrektts — COMMENTED** [review]
Thanks for the refactoring. Could you please check my comments?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (danieleades) [reply .github/workflows/ci.yml:36]: i've added `#![deny(clippy::all)]` to `lib.rs` instead. This has the same effect, but with the advantage of making the behaviour consistent between CI and local dev

File: Cargo.toml (hunk 2 above)
Comment (danieleades) [reply Cargo.toml:16]: `1.56.1` is the current MSRV as determined using [cargo-msrv](https://github.com/foresterre/cargo-msrv)

File: .github/workflows/ci.yml (hunk 3 above)
Comment (danieleades) [reply .github/workflows/ci.yml:15]: no, this one can be stable (though being nightly also does not hurt). Would you prefer me to revert this to stable?

File: .github/workflows/ci.yml (hunk 4 above)
Comment (danieleades) [reply .github/workflows/ci.yml:23]: there are occasionally some edge cases that nightly rustfmt can resolve better than stable, though in general they're fairly consistent. The toolchain used for formatting doesnt impact the stability guarantees of the crate itself, so i guess it's a matter of choice. Do you prefer using stable or nightly compiler in your local dev environment?

File: .github/workflows/ci.yml (hunk 5 above)
Comment (danieleades) [reply .github/workflows/ci.yml:33]: this one should be nightly. Clippy nightly has access to significantly more lints. The `rust-version` config option in `Cargo.toml` prevents clippy from making suggestions including syntax not supported by the MSRV, so there's no reason not to use a more recent version of clippy

File: .github/workflows/ci.yml (hunk 1 above)
Comment (danieleades) [reply .github/workflows/ci.yml:36]: i see what you mean. the 'non-snake-case' lint is a compiler warning, not a clippy warning. I've updated the catch this too.

This is now configured in the root of each target.

The question now is, is the benefit of having the compiler flag inline in the code worth the tradeoff of having to configure it in three places? I think it is, but what's your preference?

**Pull request merged**